### PR TITLE
OHRM5X-2045: Develop claim expense type screens

### PR DESCRIPTION
**Pull request by DulsaraNethmin**

### CODE REVIEW

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/SaveClaimExpenseType.vue [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        {{ $t('claim.add_expense_type') }}
```
Comment (ManulMax) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/SaveClaimExpenseType.vue:25]: In spec it shows as **Add Expense Types**. But fell like this is the correct one.

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+.orangehrm-sm-field {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: $oxd-input-control-vertical-padding 0rem;
+  &-label {
+    font-weight: 700;
+    font-size: $oxd-input-control-font-size;
+  }
+}
```
Comment (ManulMax) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue:165]: seems like repeating in claim event and expense type. Can't this move to seperate file ?

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/SaveClaimExpenseType.vue [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+<style lang="scss" scoped>
+.orangehrm-sm-field {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: $oxd-input-control-vertical-padding 0rem;
+  &-label {
+    font-weight: 700;
+    font-size: $oxd-input-control-font-size;
+  }
```
Comment (ManulMax) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/SaveClaimExpenseType.vue:162]: .

**Review by ManulMax — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmClaimPlugin/components/ExpenseTypeAutocomplete.vue [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <oxd-input-field
+    type="autocomplete"
+    :clear="false"
+    label="Name"
```
Comment (Super-Chama) [inline src/client/src/orangehrmClaimPlugin/components/ExpenseTypeAutocomplete.vue:25]: lang string

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/ClaimExpenseType.vue [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+
+const defaultSortOrder = {
+  'expenseType.name': 'ASC',
+  'expenseType.status': 'DESC',
+};
+
+export default {
+  components: {
+    'delete-confirmation': DeleteConfirmationDialog,
+    'type-autocomplete': ExpenseTypeAutocomplete,
```
Comment (Super-Chama) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/ClaimExpenseType.vue:120]:
```suggestion
    'expense-type-autocomplete': ExpenseTypeAutocomplete,
```

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        {{ $t('claim.edit_expense_type') }}
+      </oxd-text>
+
+      <oxd-divider />
+
+      <oxd-form novalidate="true" :loading="isLoading" @submit-valid="onSave">
```
Comment (Super-Chama) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue:30]:
```suggestion
      <oxd-form :loading="isLoading" @submit-valid="onSave">
```

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+<script>
+import {OxdSwitchInput} from '@ohrm/oxd';
+import {navigate} from '@ohrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import {shouldNotExceedCharLength} from '@ohrm/core/util/validation/rules';
+
+const initialExpenseTypes = {
+  name: '',
+  description: '',
+  status: null,
```
Comment (Super-Chama) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue:86]: initial status should be `false` ?

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue (hunk 7 above)
Comment (DulsaraNethmin) [reply src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue:86]: ok

File: src/client/src/orangehrmClaimPlugin/pages/claim-style.scss [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    padding: $oxd-input-control-vertical-padding 0rem;
+    &-label {
+      font-weight: 700;
+      font-size: $oxd-input-control-font-size;
+    }
+  }
+
```
Comment (Super-Chama) [inline src/client/src/orangehrmClaimPlugin/pages/claim-style.scss:27]: keep 1 new line at end remove anything else

File: src/client/src/orangehrmClaimPlugin/pages/claim-style.scss [hunk 9]
```diff
@@ -0,0 +1,27 @@
+// OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
```
Comment (Super-Chama) [inline src/client/src/orangehrmClaimPlugin/pages/claim-style.scss:1]: I prefer the name of this stylesheet be specific to a component. 
`save-claim-event.scss`

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmClaimPlugin/pages/claim-style.scss (hunk 8 above)
Comment (DulsaraNethmin) [reply src/client/src/orangehrmClaimPlugin/pages/claim-style.scss]: i added a new line. but not appear the change

**Review by Super-Chama — APPROVED** [review]

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/ClaimExpenseType.vue [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+              onClick: this.onClickEdit,
+              props: {
+                name: 'pencil-fill',
+              },
+            },
+          },
+        },
+      ],
+      checkedItems: [],
+      ExpenseTypeStatuses: [
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/ClaimExpenseType.vue:228]:
```suggestion
      expenseTypeStatuses: [
```

File: src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/ClaimExpenseType.vue [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+              v-model="filters.name"
+              :label="$t('general.name')"
+            />
+          </oxd-grid-item>
+          <oxd-grid-item>
+            <oxd-input-field
+              v-model="filters.status"
+              type="select"
+              :label="$t('general.status')"
+              :options="ExpenseTypeStatuses"
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/ClaimExpenseType.vue:37]:
```suggestion
              :options="expenseTypeStatuses"
```

**Review by RajithaKumara — APPROVED** [review]
TODO:: 

- handle screen permissions for claim
- add menu items

**Pull request merged**